app/assets/javascripts/discourse/app/initializers/post-decorations.js

```diff
@@ -108,24 +108,25 @@ export default {
 
       api.decorateCookedElement(
         (element) => {
-          const containers = element.getElementsByClassName("video-container");
-          for (let i = 0; i < containers.length; ++i) {
-            const video = containers[i].getElementsByTagName("video")[0];
-            video.addEventListener("loadeddata", () => {
-              later(() => {
-                if (video.videoWidth === 0 && video.videoHeight === 0) {
-                  const notice = document.createElement("div");
-                  notice.className = "notice";
-                  notice.innerHTML =
-                    iconHTML("exclamation-triangle") +
-                    " " +
-                    I18n.t("cannot_render_video");
+          element
+            .querySelectorAll(".video-container")
+            .forEach((videoContainer) => {
+              const video = videoContainer.getElementsByTagName("video")[0];
+              video.addEventListener("loadeddata", () => {
+                later(() => {
```

Comment:
We should check if we still have video in the view, what happens if you navigate away from the topic during the 500ms ?

try to put a longer delay like 5seconds, and you should see an error in the console

Reply:
What error do you see? I do not see any. I tried as you said, set a timeout of 5 seconds and then navigated to the homepage.

Reply:
I don't see any error, I expected there would be one, because you don't cancel the later block when navigating away:

```
- display topic
- shows warning
- navigate away from topic before later is done
- attempts to do something with video element, but not in DOM anymore: ERROR
```

well if not the case, ignore my comment, :+1: